src/SAML2/Utils.php

```diff
@@ -649,23 +650,30 @@ public static function addStrings(
     /**
      * Create a KeyDescriptor with the given certificate.
      *
-     * @param string $x509Data The certificate, as a base64-encoded DER data.
+     * @param string|null $x509Data The certificate, as a base64-encoded DER data.
+     * @param string|null $keyName The name of the key as specified in the KeyInfo
      * @return \SAML2\XML\md\KeyDescriptor The keydescriptor.
      */
-    public static function createKeyDescriptor(string $x509Data) : KeyDescriptor
+    public static function createKeyDescriptor(string $x509Data = null, string $keyName = null) : KeyDescriptor
```

Comment:
The way the signature of the function is right now, it doesn't allow the use case supported by the code of a `$keyName` being passed but no `$x509Data`, since it's typed and not nullable. So this signature allows only for three cases:

* Passing the certificate and the key name.
* Passing the certificate but no key name (backwards-compatible).
* Not passing any of them at all.

I suspect the one that's missing (passing only key name) is also desired looking at the code, so we should change the method's signature to make `$x509Data` nullable (probably both arguments, actually). If my assumption is wrong, then I guess we should change the code to make sure we always include a certificate (and the signature to remove the default value from `$x509Data`).

Reply:
I added nullable types to the signature.